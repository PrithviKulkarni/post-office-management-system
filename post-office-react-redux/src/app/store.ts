import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
