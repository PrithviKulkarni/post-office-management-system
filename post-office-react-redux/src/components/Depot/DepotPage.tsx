import { SyntheticEvent } from "react";
import { useNavigate } from "react-router-dom";
import { Button, Form } from 'react-bootstrap'
import DepotDetails from "./DepotDetails"
import UpdateDepot from "./UpdateDepot";
import DeleteDepot from "./DeleteDepot";


const DepotPage = () => {
    const navigate = useNavigate();

    const submitHandler = (event: SyntheticEvent) => {

        event.preventDefault();

        navigate('/depots');

    }

    return (
        <div data-testid="depotPage">
            <div>
                <DepotDetails />
            </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div>
                <UpdateDepot/>
            </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <DeleteDepot/>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <Form onSubmit={submitHandler}>
                <Button type="submit" className="button" data-testid="buttonO">Return</Button>
            </Form>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
            <div> &nbsp; </div>
        </div>
    )
}

export default DepotPage