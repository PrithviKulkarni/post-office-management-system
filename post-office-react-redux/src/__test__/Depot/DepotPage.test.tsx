import { render, cleanup, fireEvent } from '@testing-library/react';
import DepotPage from '../../components/Depot/DepotPage';
import { screen } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from '../../app/store';
import { BrowserRouter } from 'react-router-dom';

afterEach(cleanup);
it("Depot speific page render test", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
      <DepotPage /></BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies depots has rendered", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
      <DepotPage /></BrowserRouter>
    </Provider>
  );
  const button = getByTestId('buttonO')
  expect(button.textContent).toBe('Return')
  fireEvent.click(button)
  expect(button).toBeTruthy()
  expect(screen.getByTestId("depotPage")).toBeInTheDocument();
});