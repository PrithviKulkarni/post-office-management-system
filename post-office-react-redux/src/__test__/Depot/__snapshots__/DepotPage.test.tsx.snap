// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Depot speific page render test 1`] = `
<DocumentFragment>
  <div
    data-testid="depotPage"
  >
    <div>
      <section
        data-testid="depotDetails"
      />
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
      <div
        class="updateDepot container"
        data-testid="updateDepot"
      >
        <form
          class=""
          data-testid="create-depot-city"
        >
          <div
            class="city input-group"
          >
            <span
              class="input-group-text"
              id="basic-addon1"
            >
              Depot City
            </span>
            <input
              aria-describedby="basic-addon1"
              aria-label="city"
              class="form-control form-control-sm"
              data-testid="cityIn"
              name="city"
              type="text"
              value=""
            />
          </div>
          <div
            class="workers input-group"
            data-testid="create-driver-workers"
          >
            <span
              class="input-group-text"
              id="basic-addon1"
            >
              Number of Workers
            </span>
            <input
              aria-describedby="basic-addon1"
              aria-label="workers"
              class="form-control form-control-sm"
              data-testid="workersIn"
              name="workers"
              type="text"
              value="0"
            />
          </div>
          <button
            class="button btn btn-primary"
            data-testid="button"
            type="submit"
          >
            Update Depot Details
          </button>
          <p />
        </form>
      </div>
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div
      class="container"
    >
      <form
        class="deleteDepot"
        data-testid="deleteDepot"
      >
        <button
          class="button btn btn-primary"
          data-testid="button"
          type="submit"
        >
          Delete Depot
        </button>
      </form>
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <form
      class=""
    >
      <button
        class="button btn btn-primary"
        data-testid="buttonO"
        type="submit"
      >
        Return
      </button>
    </form>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
    <div>
         
    </div>
  </div>
</DocumentFragment>
`;
